src/core/interfaces/user.ts

```diff
@@ -0,0 +1,9 @@
+interface IName {
+    firstName: string;
+    lastName: string;
+}
+
+interface IUser extends IName {
+    email: string;
+    password: string;
+}
```

Comment:
What benefits do you see in separating these interfaces? I believe they can be merged, no?